File: daggers/src/test/java/vn/tiki/daggers/di/ActivityScope.java
package vn.tiki.daggers.di;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import javax.inject.Scope;

/**
 * Identifies a type that the injector only instantiates once. Not inherited.
 *
 * @see Scope @Scope
 */
@Scope
@Documented
@Retention(RUNTIME)
public @interface ActivityScope {

}

File: sample/src/main/java/vn/tiki/sample/di/ActivityScope.kt
package vn.tiki.sample.di

import javax.inject.Scope
import kotlin.annotation.AnnotationRetention.RUNTIME

/**
 * Identifies a type that the injector only instantiates once. Not inherited.
 *
 * @see Scope @Scope
 */
@Scope
@MustBeDocumented
@kotlin.annotation.Retention(RUNTIME)
annotation class ActivityScope
